engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoFile.java

```diff
@@ -203,9 +203,52 @@ Object getMembers(boolean includeInternal) throws UnsupportedMessageException {
       return ArrayLikeHelpers.wrapStrings(MEMBERS);
     }
 
-    @TruffleBoundary
+    @TruffleBoundary(allowInlining = true)
+    private int read() throws IOException {
+      return delegate.read();
+    }
+
+    @TruffleBoundary(allowInlining = true)
+    private byte[] readNBytes(int limit) throws IOException {
+      return delegate.readNBytes(limit);
+    }
+
+    @TruffleBoundary(allowInlining = true)
+    private ByteBuffer readNByteBuffer(int limit) throws IOException {
+      return ByteBuffer.wrap(delegate.readNBytes(limit));
+    }
+
+    @TruffleBoundary(allowInlining = true)
+    private ByteBuffer readAllBytes() throws IOException {
+      return ByteBuffer.wrap(delegate.readAllBytes());
+    }
+
+    @TruffleBoundary(allowInlining = true)
+    private void skipNBytes(long n) throws IOException {
+      delegate.skipNBytes(n);
+    }
+
+    @TruffleBoundary(allowInlining = true)
+    private boolean markSupported() throws IOException {
+      return delegate.markSupported();
```

Comment:
Don't we need to also expose `mark` and `reset` methods and pass through to the `delegate`?

Reply:
Those methods were not yet needed for anything. We can add them: 633e4d52b3

Reply:
They were being used by `Stream_Utils::peek`.